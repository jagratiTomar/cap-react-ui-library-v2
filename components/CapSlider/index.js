/**
*
* CapSlider
*
*/
import React from 'react';
import './_capSlider.scss';
import { Slider } from 'antd';
import PropTypes from 'prop-types';
const classNames = require('classnames');

class CapSlider extends React.Component { // eslint-disable-line react/prefer-stateless-function
  translateType(type) {
    let tipFormatter = null;
    switch (type) {
      case 'ratio':
        tipFormatter = (value) => `${value}|${100 - value}`;
        break;
      default:
        tipFormatter = null;
    }
    return tipFormatter;
  }

  render() {
    const {type, ...rest} = this.props;
    const formatter = this.translateType(type);
    return (
      <div>
        <Slider tipFormatter={formatter} {...rest} className={classNames("cap-slider", this.props.className)}>
          {React.Children.toArray(this.props.children)}
        </Slider>
      </div>
    );
  }
}

CapSlider.propTypes = {
  children: PropTypes.any,
  className: PropTypes.string,
};

export default CapSlider;
