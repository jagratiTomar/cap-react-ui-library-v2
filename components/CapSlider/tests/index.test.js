import CapSlider  from '../index';
import renderer from 'react-test-renderer';
import React from 'react'; // eslint-disable-line no-unused-vars

describe('<CapSlider />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CapSlider />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

