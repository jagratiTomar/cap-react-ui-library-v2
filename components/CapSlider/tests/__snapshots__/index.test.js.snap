// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapSlider /> renders correctly 1`] = `
<div>
  <div
    className="ant-slider cap-slider"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="ant-slider-rail"
      style={Object {}}
    />
    <div
      className="ant-slider-track"
      style={
        Object {
          "left": "0%",
          "width": "0%",
        }
      }
    />
    <div
      className="ant-slider-step"
    />
    <div
      aria-disabled={false}
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="ant-slider-handle"
      onBlur={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="slider"
      style={
        Object {
          "left": "0%",
        }
      }
      tabIndex={0}
    />
    <div
      className="ant-slider-mark"
    />
  </div>
</div>
`;
